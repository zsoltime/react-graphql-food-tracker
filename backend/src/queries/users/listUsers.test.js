'use strict';

const listUsers = require('./listUsers');

describe('listUsers query', () => {
  it('lists users', async () => {
    const mockedContext = {
      db: {
        query: {
          user: jest
            .fn()
            .mockImplementation(() => ({ permissions: ['ADMIN'] })),
          users: jest.fn().mockImplementation(() => []),
        },
      },
      request: { userId: 'testUserId' },
    };

    await listUsers(null, null, mockedContext, 'testInfoObject');

    expect(mockedContext.db.query.users).toHaveBeenCalled();
    expect(
      mockedContext.db.query.users.mock.calls[0]
    ).toMatchSnapshot();
  });

  it('throws error if user is not logged in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      listUsers(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error if user does not have permssion', () => {
    const mockedContext = {
      db: {
        query: {
          user: jest
            .fn()
            .mockImplementation(() => ({ permissions: ['USER'] })),
        },
      },
      request: { userId: 'testUserId' },
    };

    expect(
      listUsers(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
