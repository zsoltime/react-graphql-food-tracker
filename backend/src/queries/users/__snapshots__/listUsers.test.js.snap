// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listUsers query lists users 1`] = `
Array [
  Object {},
  "testInfoObject",
]
`;

exports[`listUsers query throws error if user does not have permssion 1`] = `"You don't have permission"`;

exports[`listUsers query throws error if user is not logged in 1`] = `"You must be signed in to list users"`;
